src/java/org/apache/cassandra/tcm/log/LogReader.java

```diff
@@ -148,7 +148,8 @@ else if (includeSnapshot)
             else if (closestSnapshot.epoch.isBefore(start))
             {
                 ImmutableList.Builder<Entry> entries = new ImmutableList.Builder<>();
-                EntryHolder entryHolder = getEntries(closestSnapshot.epoch.nextEpoch(), end);
+                // start is exclusive, so use the closest snapshot
+                EntryHolder entryHolder = getEntries(closestSnapshot.epoch, end);
```

Comment:
> 8) TCM’s getLogEntries method hit an edge case with snapshots where it assumed the API was inclusive, but its exclusive; this caused a gap in epochs

Reply:
Oh, I have missed this one looks like. Wondering if all the tests are still passing.